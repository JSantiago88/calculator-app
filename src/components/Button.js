import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, styles, handleSelectButton }) => {

    return (
        <>
            <button
                className={styles}
                onClick={handleSelectButton}
            >
                {value}
            </button>
        </>
    )
};

Button.propTypes = {
    handleSelectButton: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired,
    styles: PropTypes.string.isRequired
}

export default Button;