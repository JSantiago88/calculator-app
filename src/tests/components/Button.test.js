import React from 'react';
import { shallow } from 'enzyme';

import Button from '../../components/Button';

describe('Test in <Button />', () => {

    const handleSelectButton = () => {};
    const value = '5';
    const styles = 'btn';

    const wrapper = shallow(<Button
        styles={styles}
        value={value}
        handleSelectButton={handleSelectButton}
    />);

    test('should Button the component correctly', () => {

        expect(wrapper).toMatchSnapshot();
    });


    test('should have a button label', () => {
        
        const button = wrapper.find('button');

        expect(button.prop('onClick')).not.toBe(undefined);
    });
    

    test('should have a text in the button', () => {
        const button = wrapper.find('button');

        expect(button.text().trim()).toBe(value);
    });
});
