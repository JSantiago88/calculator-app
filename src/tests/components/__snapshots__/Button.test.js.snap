// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Button /> should Button the component correctly 1`] = `
<Fragment>
  <button
    className="btn"
    onClick={[Function]}
  >
    5
  </button>
</Fragment>
`;
